### DISCUSSION TITLE
contrib sub-layer obsolete ?

### DISCUSSION THREAD

**Discussion by andregirard37** [OE4T/meta-tegra#1253] [Q&A] [answered]
Hello,
I am updating my meta-tegra from Gategarth - JetPack 4.4.1, to Mickledore - JetPack 5.1.1 and it appears that meta-tegra/contrib is gone ? Has there been a refactoring of the code or is it gone for good ? I was using the gpu-burn tool for example, that I can't find anymore.

Thanks!

**Comment by madisongh** [comment]
The main reasons for the contrib layer - the gcc 8 toolchain and recipes for older gstreamer plugins - have gone away, so it was retired.  We have a [separate repo](https://github.com/OE4T/meta-tegra-community) for community-maintained recipes, which was the other reason it was there.

As for gpu-burn, it was getting problematic to keep it working properly for the Jetsons, so I didn't migrate the recipe for it over to the community repo.

**Reply by andregirard37** [reply]
Thanks for the quick reply!
Are you aware of any replacement to stress test the GPU/CPU ?

**Comment by ichergui** [comment]
Hey @andregirard37 

You can try [GPUStressTest](https://github.com/NVIDIA/GPUStressTest) project. I'm using it for stressing the dGPU

**Reply by dc-buettgenbach** [reply]
@madisongh : we are currently trying to build an image for Jetson with yocto (using kirkstone) and need to add gstreamer. Our current application uses 1.14 since this is the preferred version in L4T. 
gestream 1.14 was part of the contrib layer before. And in meta-tegra-community there is no recipe for gstreamer at all. So how to install gstreamer 1.14 for kirkstone smoothly? Poky ships with 1.20.x and we are afraid that might break our application or lead to other side effects. Is there a maintained solution?

Thanky for any reply!

**Reply by dwalkes** [reply]
Hi @dc-buettgenbach I suspect you are looking for the kirkstone-l4t-r32.7.x branch at https://github.com/OE4T/meta-tegra/tree/kirkstone-l4t-r32.7.x/contrib/recipes-multimedia/gstreamer-1.14.

The kirkstone branch is Jetpack 5, which uses a later gstreamer even in unmodified Nvidia source I believe.

See https://github.com/OE4T/meta-tegra/wiki/Which-branch-should-I-use%3F

**Reply by dc-buettgenbach** [reply]
Hello @dwalkes @madisongh,

Thank you for your prompt and insightful response.

I would like to confirm that we are currently utilizing Jetpack 4.6 on our Jetson boards, and the branch you've pointed out aligns with our setup.

Regarding Yocto, we have successfully established a basic image using the Kirkstone version. Our objective is to ensure compatibility with Jetpack 5.1.2. However, as mentioned the 'contrib' folder appears to be absent in Kirkstone, and there seems to be a lack of GStreamer recipes within the meta-tegra or meta-tegra-community layers. Presently, the recipes available are primarily for GStreamer plugins, while GStreamer itself is being compiled through Poky, which includes version 1.20.x.

This leads to a concern as Jetpack 5.1.2 (L4T 35.4.1) is bundled with GStreamer 1.16. We are curious about the absence of GStreamer 1.16 recipes in meta-tegra. Furthermore, would you recommend proceeding with GStreamer, and if so, should we consider maintaining our own recipe for GStreamer 1.16, although we'd prefer to avoid this?

Additionally, we are facing an issue with DeepStream, which has dependencies on X11. Given the extensive nature of the X11 package, we are keen to exclude it from our image. Could you possibly offer any guidance or alternative solutions for this situation?

Your expertise and any assistance you can provide would be greatly appreciated.

**Reply by madisongh** [reply]
Starting with the JetPack 5-based branches, the gstreamer recipes in the contrib layer were obsoleted, as they were mainly there to provide compatibility for NVIDIA containers.  With the JP5 branches, the gstreamer plugins are compatible with the current OE-Core gstreamer, and we have a different mechanism for providing compatibility with NVIDIA containers, via the tegra-container-passthrough recipe.

**Reply by dwalkes** [reply]
> Additionally, we are facing an issue with DeepStream, which has dependencies on X11. Given the extensive nature of the X11 package, we are keen to exclude it from our image. Could you possibly offer any guidance or alternative solutions for this situation?

See https://github.com/OE4T/meta-tegra/issues/1461#issuecomment-1902078923 for a recent discussion about this.

**Reply by dc-buettgenbach** [reply, marked as answer]
@madisongh Thank you for the clarification regarding the GStreamer recipes within the JetPack 5-based branches and the transition to a new compatibility mechanism via the tegra-container-passthrough recipe. Based on your guidance, we have decided to proceed with using Kirkstone and the current OE-Core GStreamer version.

@dwalkes Regarding DeepStream, I will continue the discussion in issue #1461 as suggested.

I appreciate your assistance throughout this process. Please feel free to close the issue at your convenience.
